test/cli_arguments_test.dart

```diff
@@ -0,0 +1,229 @@
+          expect(await _resolve(r'"\""'), equals(['"']));
+        });
+
+        test("an escaped backslash", () async {
+          expect(await _resolve(r'"\\"'), equals(["\\"]));
+        });
+      });
+
+      group("with single quotes containing", () {
+        test("nothing", () async {
+          expect(await _resolve("''"), equals([""]));
+        });
+
+        test("spaces", () async {
+          expect(await _resolve("' foo bar '"), equals([" foo bar "]));
+        });
+
+        test("a double quote", () async {
+          expect(await _resolve("'\"'"), equals(['"']));
+        });
+
+        test("an escaped single quote", () async {
+          expect(await _resolve(r"'\''"), equals(["'"]));
+        });
+
+        test("an escaped backslash", () async {
+          expect(await _resolve(r"'\\'"), equals(["\\"]));
+        });
+      });
+
+      test("with plain text adjacent to quotes", () async {
+        expect(await _resolve("\"foo bar\"baz'bip bop'"),
+            equals(["foo barbazbip bop"]));
+      });
+    });
+
+    group("multiple arguments", () {
+      test("separated by single spaces", () async {
+        expect(await _resolve("a b c d"), equals(["a", "b", "c", "d"]));
+      });
+
+      test("separated by multiple spaces", () async {
+        expect(await _resolve("a  b   c    d  "), equals(["a", "b", "c", "d"]));
+      });
+
+      test("with different quoting styles", () async {
+        expect(await _resolve("a \"b c\" 'd e' f\\ g"),
+            equals(["a", "b c", "d e", "f g"]));
+      });
+    });
+  });
+
+  group("globbing", () {
+    test("lists files that match the glob", () async {
+      await d.file("foo.txt").create();
+      await d.file("bar.txt").create();
+      await d.file("baz.zip").create();
+
+      expect(await _resolve("ls *.txt"), equals(["ls", "foo.txt", "bar.txt"]));
+    });
+
+    test("an absolute glob expands to absolute paths", () async {
+      await d.file("foo.txt").create();
+      await d.file("bar.txt").create();
+      await d.file("baz.zip").create();
+
+      var glob = p.join(Glob.quote(d.sandbox), "*.txt");
+      expect(await _resolve("ls $glob"),
+          equals(["ls", d.path("foo.txt"), d.path("bar.txt")]));
+    });
+
+    test("ignores glob characters in quotes", () async {
+      await d.file("foo.txt").create();
+      await d.file("bar.txt").create();
+      await d.file("baz.zip").create();
+      expect(await _resolve("ls '*.txt'"), equals(["ls", "*.txt"]));
+    });
+
+    test("ignores backslash-escaped glob characters", () async {
+      await d.file("foo.txt").create();
+      await d.file("bar.txt").create();
+      await d.file("baz.zip").create();
+      expect(await _resolve(r"ls \*.txt"), equals(["ls", "*.txt"]));
+    });
+
+    test("returns plain strings for globs that don't match", () async {
+      expect(await _resolve("ls *.txt"), equals(["ls", "*.txt"]));
+    });
+  }, testOn: "!windows");
```

Comment:
Consider adding a test that only runs on Windows that ensures globs aren't resolved.